src/raw.rs

```diff
@@ -0,0 +1,84 @@
+#![allow(non_camel_case_types)]
+
+use pgx::*;
+
+// TODO: Is this the right approach to declaring `bytea` and `TimestampTz`?
+extension_sql!("",
+    name = "pseudo_create_types",
+    creates = [Type(bytea), Type(TimestampTz),],
+    bootstrap,
+);
+
+macro_rules! raw_type {
+    ($name:ident, $tyid: path, $arrayid: path) => {
+        impl FromDatum for $name {
+            const NEEDS_TYPID: bool = false;
+
+            unsafe fn from_datum(
+                datum: pg_sys::Datum,
+                is_null: bool,
+                _typoid: pg_sys::Oid,
+            ) -> Option<Self>
+            where
+                Self: Sized,
+            {
+                if is_null {
+                    return None;
+                }
+                Some(Self(datum))
+            }
+        }
+
+        impl IntoDatum for $name {
+            fn into_datum(self) -> Option<pg_sys::Datum> {
+                Some(self.0)
+            }
+            fn type_oid() -> pg_sys::Oid {
+                $tyid
+            }
+            fn array_type_oid() -> pg_sys::Oid {
+                $arrayid
+            }
+        }
+
+        impl From<pg_sys::Datum> for $name {
+            fn from(d: pg_sys::Datum) -> Self {
+                Self(d)
+            }
+        }
+
+        impl Into<pg_sys::Datum> for $name {
+            fn into(self) -> pg_sys::Datum {
+                self.0
+            }
+        }
+    };
+}
+
+#[derive(Clone, Copy)]
+pub struct bytea(pub pg_sys::Datum);
+
+raw_type!(bytea, pg_sys::BYTEAOID, pg_sys::BYTEAARRAYOID);
+
+/// We define our own `TimestampTz` data type which wraps and has interop with the underlying \
```

Comment:
Honestly I'm going to claim innocence here, this is code that I copied from toolkit.

I believe that they're here for schema generation. I'm not sure if there are suitable alternatives, and it's something that I wanted to investigate once the schema generation works cleanly (after the follow-on changes). From what I understand, `pgx` does offer a `TimestampWithTimeZone` type which _might_ be an alternative to our `TimestampTz`, although from the underlying datatypes it doesn't seem to be _quite_ the same thing. It may also offer an alternative to our `bytea` type.

@JLockerman can you perhaps shed some light on this?